### PR TITLE
Add products to cart

### PR DESCRIPTION
**Pull request by Svitlana-G**

### CODE REVIEW

File: frontend/src/components/Layouts/Header/Header.module.scss [hunk 1]
```diff
@@ -103,10 +103,24 @@ header {
...
 }
 .btnCatalogue {
   position: relative;
 }
+.cartQuantity {
+  height: 14px;
+  width: 14px;
+  border-radius: 50%;
+  background-color: #de5f38;
+  color: white;
```
Comment (annaBalabak) [inline frontend/src/components/Layouts/Header/Header.module.scss:116]: use variable

File: frontend/src/components/Layouts/Header/components/BurgerMenu/BurgerMenu.module.scss [hunk 2]
```diff
@@ -89,6 +89,18 @@
 .open .menuItems {
   display: block;
 }
+.cartQuantity {
+  height: 14px;
+  width: 14px;
+  border-radius: 50%;
+  background-color: #de5f38;
```
Comment (annaBalabak) [inline frontend/src/components/Layouts/Header/components/BurgerMenu/BurgerMenu.module.scss:96]: use variable

File: frontend/src/components/Layouts/Header/components/BurgerMenu/BurgerMenu.module.scss [hunk 3]
```diff
@@ -89,6 +89,18 @@
...
+  color: white;
```
Comment (annaBalabak) [inline frontend/src/components/Layouts/Header/components/BurgerMenu/BurgerMenu.module.scss:97]: use variable

File: frontend/src/components/common/Cart/CartPopup.module.scss [hunk 4]
```diff
@@ -57,7 +52,7 @@
 
 @include sm {
   .container {
-    padding: 16px;
+    padding: 0px;
     overflow: unset;
   }
   .title {
```
Comment (annaBalabak) [inline frontend/src/components/common/Cart/CartPopup.module.scss:58]: Should be centered

File: frontend/src/components/common/Cart/components/CartEmpty/CartEmpty.module.scss [hunk 5]
```diff
@@ -2,18 +2,18 @@
...
 
 .root {
-  padding-top: 25px;
+  padding-top: 24px;
+  padding-bottom: 35px;
 
   p {
     color: $gray-dark-color;
     line-height: 20.8px;
+    text-align: center;
```
[the remark this answers is not in the document]
Comment (Svitlana-G) [reply frontend/src/components/common/Cart/components/CartEmpty/CartEmpty.module.scss:11]: I need   `text-align: center ` to put the text _"У кошику не має товарів"_ in the center

File: frontend/src/components/Layouts/Header/Header.module.scss [hunk 6]
```diff
@@ -103,10 +103,24 @@ header {
...
+  display: flex;
 }
 .btnCatalogue {
   position: relative;
 }
+.cartQuantity {
+  height: 14px;
+  width: 14px;
+  border-radius: 50%;
+  background-color: #de5f38;
```
[the remark this answers is not in the document]
Comment (Svitlana-G) [reply frontend/src/components/Layouts/Header/Header.module.scss:115]: Done

File: frontend/src/components/Layouts/Header/Header.module.scss (hunk 1 above)
Comment (Svitlana-G) [reply frontend/src/components/Layouts/Header/Header.module.scss:116]: Done

File: frontend/src/components/Layouts/Header/components/BurgerMenu/BurgerMenu.module.scss (hunk 2 above)
Comment (Svitlana-G) [reply frontend/src/components/Layouts/Header/components/BurgerMenu/BurgerMenu.module.scss:96]: Done

File: frontend/src/components/Layouts/Header/components/BurgerMenu/BurgerMenu.module.scss (hunk 3 above)
Comment (Svitlana-G) [reply frontend/src/components/Layouts/Header/components/BurgerMenu/BurgerMenu.module.scss:97]: Done

File: frontend/src/components/Layouts/Header/Header.jsx [hunk 7]
```diff
@@ -14,19 +14,29 @@ import { useModalEffect } from '../../../hooks/useModalEffect';
...
+import { useSelector } from 'react-redux';
+// import { calculateTotal } from '../../../redux/slices/cartSlice';
 
 const Header = () => {
   const [sticky, setSticky] = useState(false);
   const [isCatalogOpened, setIsCatalogOpened] = useState(false);
 
-  const cart = useSelector((state) => state.cartSliceReducer);
-  const dispatch = useDispatch();
+  const cartItems = useSelector((state) => state.cartSliceReducer.cartItems);
```
[the remark this answers is not in the document]
Comment (Svitlana-G) [reply frontend/src/components/Layouts/Header/Header.jsx:23]: It doesn't work

File: frontend/src/components/Layouts/Header/Header.jsx [hunk 8]
```diff
@@ -14,19 +14,29 @@ import { useModalEffect } from '../../../hooks/useModalEffect';
...
-    dispatch(calculateTotal());
-  }, [cart, dispatch]);
+  // useEffect(() => {
+  //   dispatch(calculateTotal());
+  //   console.log(cart.cartTotalQuantity);
+  // }, [cart, dispatch]);
+
+  const productQuantity = useMemo(() => {
+    let quantity = 0;
+    cartItems.forEach((item) => {
```
[the remark this answers is not in the document]
Comment (Svitlana-G) [reply frontend/src/components/Layouts/Header/Header.jsx]: Done

File: frontend/src/components/Layouts/Header/components/BurgerMenu/ClosedMenu.jsx [hunk 9]
```diff
@@ -25,6 +30,9 @@ const ClosedMenu = ({ toggleMenu, toggleCart, toggleMyBouquet }) => {
           </div>
           <div className={styles.cart}>
             <IconButton icon={<ICONS.CartIcon />} onClick={toggleCart} />
+            {cartQuantity != 0 ? (
```
Comment (halynarydasova) [inline frontend/src/components/Layouts/Header/components/BurgerMenu/ClosedMenu.jsx:33]: It's better to use strict equality

File: frontend/src/components/Layouts/Header/Header.jsx [hunk 10]
```diff
@@ -119,6 +135,9 @@ const Header = () => {
 
             <div className={styles.cart}>
               <IconButton onClick={toggleCart} icon={<ICONS.CartIcon />} />
+              {productQuantity != 0 ? (
```
Comment (halynarydasova) [inline frontend/src/components/Layouts/Header/Header.jsx:138]: It's better to use strict equality

File: frontend/src/components/Layouts/Header/Header.jsx [hunk 11]
```diff
@@ -14,11 +14,23 @@ import { useModalEffect } from '../../../hooks/useModalEffect';
...
+import { useSelector } from 'react-redux';
 
 const Header = () => {
   const [sticky, setSticky] = useState(false);
   const [isCatalogOpened, setIsCatalogOpened] = useState(false);
 
+  const cartItems = useSelector((state) => state.cartSliceReducer.cartItems);
+
+  const productQuantity = useMemo(() => {
+    let InitialValue = 0;
```
Comment (halynarydasova) [inline frontend/src/components/Layouts/Header/Header.jsx:26]: You don't need this variable

File: frontend/src/components/Layouts/Header/Header.jsx [hunk 12]
```diff
@@ -14,11 +14,23 @@ import { useModalEffect } from '../../../hooks/useModalEffect';
...
+    const quantity = cartItems.reduce(
+      (accumulator, item) => accumulator + item.cardQuantity,
+      InitialValue,
```
Comment (halynarydasova) [inline frontend/src/components/Layouts/Header/Header.jsx:29]: You can pass here 0

File: frontend/src/components/common/Cart/CartPopup.jsx [hunk 13]
```diff
@@ -9,60 +9,16 @@ import IconButton from '../../ui-kit/components/IconButton';
...
-    oldPrice: 2000,
-    actualPrice: 1500,
-  },
-];
+import { calculateTotal } from '../../../redux/slices/cartSlice';
 
 const CartPopup = ({ toggleCart }) => {
-  const totalSum = items.reduce((sum, obj) => sum + obj.actualPrice, 0);
+  const cart = useSelector((state) => state.cartSliceReducer.cartItems);
+  const total = useSelector((state) => state.cartSliceReducer.cartTotalAmount);
```
Comment (halynarydasova) [inline frontend/src/components/common/Cart/CartPopup.jsx:16]: You don't need redux here.

File: frontend/src/components/Layouts/Header/components/BurgerMenu/ClosedMenu.jsx (hunk 9 above)
Comment (Svitlana-G) [reply frontend/src/components/Layouts/Header/components/BurgerMenu/ClosedMenu.jsx:33]: Done

File: frontend/src/components/Layouts/Header/Header.jsx (hunk 10 above)
Comment (Svitlana-G) [reply frontend/src/components/Layouts/Header/Header.jsx:138]: Done

File: frontend/src/components/Layouts/Header/Header.jsx (hunk 11 above)
Comment (Svitlana-G) [reply frontend/src/components/Layouts/Header/Header.jsx:26]: Deleted

File: frontend/src/components/Layouts/Header/Header.jsx (hunk 12 above)
Comment (Svitlana-G) [reply frontend/src/components/Layouts/Header/Header.jsx:29]: Replaced by 0

File: frontend/src/components/common/Cart/CartPopup.jsx (hunk 13 above)
Comment (Svitlana-G) [reply frontend/src/components/common/Cart/CartPopup.jsx]: Deleted the function calculateTotal from redux. Added a function for calculating total amount to the react component.

File: frontend/src/components/common/Cart/CartPopup.jsx [hunk 14]
```diff
@@ -10,59 +10,16 @@ import IconButton from '../../ui-kit/components/IconButton';
...
-  },
-];
-
 const CartPopup = ({ toggleCart }) => {
-  const totalSum = items.reduce((sum, obj) => sum + obj.actualPrice, 0);
+  const cartItems = useSelector((state) => state.cartSliceReducer.cartItems);
+
+  const productTotal = useMemo(() => {
+    let total = 0;
+    cartItems.forEach((element) => {
```
Comment (halynarydasova) [inline frontend/src/components/common/Cart/CartPopup.jsx:18]: Use here reduce function as well.

**Pull request merged**